plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkExtension.kt

```diff
@@ -9,7 +10,39 @@ open class JLinkExtension
 internal constructor(
         project: Project
 ) {
-    val configure: NamedDomainObjectCollection<JLinkOptions> = project.container(JLinkOptions::class.java) { name ->
+    val configure: NamedDomainObjectContainer<JLinkOptions> = project.container(JLinkOptions::class.java) { name ->
         JLinkOptions(name = name)
     }
-}
\ No newline at end of file
+
+    /**
+     * Registers a new jlink configuration.
+     *
+     * ```
+     * jlink {
+     *   "example" {
+     *     // options...
+     *   }
+     * }
+     * ```
+     *
+     * is equivalent to
+     * ```
+     * jlink {
+     *   register("example") {
+     *     // options...
+     *   }
+     * }
+     * ```
+     */
+    operator fun String.invoke(configurationAction: Action<in JLinkOptions>) = register(this, configurationAction)
```

Comment:
Do we really want to have this method in our public API. It won't be usable from a Groovy DSL build script.

Reply:
But it _is_ usable from a kotlin build script, and makes the kotlin configuration easier IMO:

```kotlin
jlink {
  "configuration name" {
    // configuration settings
  }
}
```

versus

```kotlin
jlink {
  register("configuration name") {
    // configuration settings
  }
}
```

Reply:
We should be exposing the `NamedDomainObjectContainer`'s API as the primary way for adding `JLinkOptions`. The `NamedDomainObjectContainer` is a standard API so it's better to just let users use that without providing helper methods for interacting with it. IMO.